ouroboros-network/src/Ouroboros/Network/PeerSelection/Governor.hs

```diff
@@ -634,15 +658,35 @@ peerChurnGovernor inRng base peerSelectionVar = do
     -- Randomly delay between churnInterval and churnInterval + maxFuzz seconds.
     fuzzyDelay :: StdGen -> Time -> m StdGen
     fuzzyDelay rng execTime = do
-      let (fuzz, rng') = randomR (0, 600 :: Double) rng
-      threadDelay $ (realToFrac fuzz) + (diffTime churnInterval execTime)
+      mode <- atomically getFetchMode
+      case mode of
+           FetchModeDeadline -> longDelay rng execTime
+           FetchModeBulkSync -> shortDelay rng execTime
+
+    fuzzyDelay' :: Time -> Double -> StdGen -> Time -> m StdGen
+    fuzzyDelay' baseDelay maxFuzz rng execTime = do
+      let (fuzz, rng') = randomR (0, maxFuzz :: Double) rng
+          !delay = realToFrac fuzz + diffTime baseDelay execTime
+      traceWith tracer $ TraceChurnWait delay
+      threadDelay delay
       return rng'
 
+
+    longDelay :: StdGen -> Time -> m StdGen
+    longDelay = fuzzyDelay' churnInterval 600
+
+
+    shortDelay :: StdGen -> Time -> m StdGen
+    shortDelay = fuzzyDelay' churnIntervalBulk 60
+
     -- The min time between running the churn governor.
     churnInterval :: Time
     churnInterval = Time 3300
 
 
+    churnIntervalBulk :: Time
```

Comment:
This ought to be `DiffTime`, as this expresses period of time.  This will influence `fuzzyDelay'`.

Reply:
Done